src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,151 @@
+package duke.task;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import duke.Storage;
+import duke.Ui;
+import duke.exception.DukeException;
+
+/**
+ * A class contains the task list and deal with the command run on the task list.
+ */
+public class TaskList {
+
+    private List<Task> taskList;
+
+    /**
+     * Constructs a new TaskList containing a empty task list.
+     */
+    public TaskList() {
+        this(new ArrayList<>());
+    }
+
+    /**
+     * Constructs a new TaskList containing a given task list.
+     *
+     * @param taskList the given List of task.
+     */
+    public TaskList(List<Task> taskList) {
+        this.taskList = taskList;
+    }
+
+    /**
+     * Executes the command on the list of tasks.
+     *
+     * @param commands an Array of String which contains command and its arguments.
+     * @param ui       the Ui to deal with the interactions with users.
+     * @param storage  the Storage to deals with loading tasks from the file and saving tasks in the file.
+     * @throws DukeException thrown if the storage process goes wrong.
+     */
+    public void runCommand(String[] commands, Ui ui, Storage storage) throws DukeException {
```

Comment:
Cool idea of placing the runCommand function in the TaskList clear. An alternative could be placing this function in the respective Command classes!